### PR TITLE
LLM: add `optimize_model` examples for llama2 and chatglm 

### PR DESCRIPTION
**Pull request by plusbang**
## Description

After previous validation, add `optimize_model` examples for llama2 and chatglm 

### 3. Summary of the change 

Add python scripts and readme

### 4. How to test?
- [ ] Unit test
- [x] Local test

### CODE REVIEW

File: python/llm/example/transformers/general_int4/chatglm/generate.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                        help='Max tokens to predict')
+
+    args = parser.parse_args()
+    model_path = args.repo_id_or_model_path
+
+    # Load model
+    model = AutoModel.from_pretrained(model_path, trust_remote_code=True)
+
+    # With only one line to enable BigDL-LLM optimization on model
+    model = optimize_model(model, low_bit="sym_int4", optimize_llm=True)
```
Comment (jason-dai) [inline python/llm/example/transformers/general_int4/chatglm/generate.py:45]: `optimize_llm` should default to true

File: python/llm/example/transformers/general_int4/llama2/generate.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                        help='Max tokens to predict')
+
+    args = parser.parse_args()
+    model_path = args.repo_id_or_model_path
+
+    # Load model
+    model = AutoModelForCausalLM.from_pretrained(model_path, trust_remote_code=True)
+    
+    # With only one line to enable BigDL-LLM optimization on model
+    model = optimize_model(model, low_bit="sym_int4", optimize_llm=True)
```
[the remark this answers is not in the document]
Comment (plusbang) [reply python/llm/example/transformers/general_int4/llama2/generate.py]: > `low_bit` should default to "sym_int4", `optimize_llm` should default to true

Have updated the default usage.

File: python/llm/example/transformers/general_int4/llama2/README.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+```bash
+conda create -n llm python=3.9
+conda activate llm
+
+pip install --pre --upgrade bigdl-llm[all] # install the latest bigdl-llm nightly build with 'all' option
+```
+
+### 2. Run
+```
+python ./generate.py --repo-id-or-model-path REPO_ID_OR_MODEL_PATH --prompt PROMPT --n-predict N_PREDICT
```
Comment (shane-huang) [inline python/llm/example/transformers/general_int4/llama2/README.md:20]: maybe change to command that user can directly copy/paste as we discussed before.

File: python/llm/example/transformers/general_int4/chatglm/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+```bash
+conda create -n llm python=3.9
+conda activate llm
+
+pip install --pre --upgrade bigdl-llm[all] # install the latest bigdl-llm nightly build with 'all' option
+```
+
+### 2. Run
+```
+python ./generate.py --repo-id-or-model-path REPO_ID_OR_MODEL_PATH --prompt PROMPT --n-predict N_PREDICT
```
[the remark this answers is not in the document]
Comment (plusbang) [reply python/llm/example/transformers/general_int4/chatglm/README.md]: > maybe change to command that user can directly copy/paste as we discussed before.

Have updated the command and related descriptions.

File: python/llm/example/transformers/general_int4/README.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# BigDL-LLM General INT4 Optimization for Large Language Model
+You can use BigDL-LLM to run general pytorch models (for example, Huggingface Transformer models) with INT4 optimizations on either servers or laptops. This directory contains example scripts to help you quickly get started using BigDL-LLM to run some popular open-source models in the community. Each model has its own dedicated folder, where you can find detailed instructions on how to install and run it.
+
+# Verified models
+| Model     | Example                                                  |
+|-----------|----------------------------------------------------------|
+| LLaMA 2   | [link](llama2)    |
+| ChatGLM   | [link](chatglm)   |
```
[the remark this answers is not in the document]
Comment (shane-huang) [reply python/llm/example/transformers/general_int4/README.md:8]: we can merge this PR first and then open another PR to put whisper under this folder.

File: python/llm/example/transformers/general_int4/README.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# BigDL-LLM General INT4 Optimization for Large Language Model
+You can use BigDL-LLM to run general pytorch models (for example, Huggingface Transformer models) with INT4 optimizations on either servers or laptops. This directory contains example scripts to help you quickly get started using BigDL-LLM to run some popular open-source models in the community. Each model has its own dedicated folder, where you can find detailed instructions on how to install and run it.
```
Comment (shane-huang) [inline python/llm/example/transformers/general_int4/README.md:2]: You can use `optimize_model` API to accelerate general PyTorch models on Intel servers and PCs.

File: python/llm/example/transformers/general_int4/README.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# BigDL-LLM General INT4 Optimization for Large Language Model
```
Comment (jason-dai) [inline python/llm/example/transformers/general_int4/README.md:1]: move to `llm/example/pytorch`

Comment (jason-dai) [reply python/llm/example/transformers/general_int4/README.md:1]: `General INT4` is confusing

File: python/llm/example/transformers/general_int4/README.md (hunk 6 above)
Comment (plusbang) [reply python/llm/example/transformers/general_int4/README.md]: > You can use `optimize_model` API to accelerate general PyTorch models on Intel servers and PCs.

Have updated. And we may need to update readme of whisper after merging this PR.

File: python/llm/example/transformers/general_int4/README.md (hunk 7 above)
Comment (plusbang) [reply python/llm/example/transformers/general_int4/README.md]: > move to `llm/example/pytorch`

Have moved and removed all `General INT4` descriptions.

**Comment by jason-dai** [comment]
change to `python/llm/example/pytorch-model`

**Comment by plusbang** [comment]
> change to `python/llm/example/pytorch-models`

Have changed the folder name.

**Comment by shane-huang** [comment]
merge now for user's urgent request. revise docs or others later.

**Pull request merged**